src/schema.ts

```diff
@@ -55,20 +80,44 @@ export const resolvers = {
 
       user.password = await bcrypt.hash(user.password, 10);
 
+      const userRepository = await getRepository(User);
+      const findUser = await userRepository.find({ email: user.email });
+      if (findUser.length > 0) {
+        throw new InputError(errorMessage.email);
+      }
+
       try {
         await getManager().save(user);
-      } catch (error) {
-        const userRepository = await getRepository(User);
-        const findUser = await userRepository.find({ email: user.email });
-
-        if (findUser.length > 0) {
-          throw new InputError(errorMessage.email, error.detail);
-        } else {
-          throw new InternalError();
-        }
+      } catch {
+        throw new InternalError();
       }
 
       return user;
     },
+
+    login: async (_, login: { data: LoginInput }): Promise<AuthPayload> => {
+      const userRepository = await getRepository(User);
+
+      const db_user = await userRepository.findOne({ email: login.data.email });
+      if (!db_user) {
+        throw new AuthError();
+      }
+
+      const valid = await bcrypt.compare(login.data.password, db_user.password);
+      if (!valid) {
```

Comment:
```suggestion
      const isPasswordValid = await bcrypt.compare(login.data.password, db_user.password);
      if (!isPasswordValid) {
```

A nomenclatura de booleans com esse sufixo `is` pode ajudar na leitura.